### ISSUE TITLE
Ubuntu 22.04 installs podman v3.4.4: Errors were encountered while processing golang-github-containers-image and golang-github-containers-common

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#1571]
Tracking: https://github.com/containers/podman/issues/14367

Installing podman on **Ubuntu 22.04** via **apt-get** fails with:

```
Errors were encountered while processing:
 /tmp/apt-dpkg-install-lvJTgR/01-golang-github-containers-image_5.16.0-3_all.deb
 /tmp/apt-dpkg-install-lvJTgR/02-golang-github-containers-common_0.44.4+ds1-1_all.deb
needrestart is being skipped since dpkg has failed
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Important:** Podman v3.4.4 is being installed here not the latest **v4.1.0** build.  Perhaps this is the underlying problem?

**Comment by jefflill** [comment]
The podman closed this issue and referred me to the Debian package managers.  I'm not quite sure how to submit an issue there and I need to move on.

The problem seems to be that two google container related dependencies are trying to update the same file (AKA _overlapping files_).  I'm going to try using the -o Dpkg::Options::="--force-overwrite"` option as described here:

https://github.com/actions/virtual-environments/issues/2703#issuecomment-778433345

**Comment by jefflill** [comment]
We're seeing the same problem installing **kubeadm**

**Comment by jefflill** [comment]
The `--force-overwrite` hack seems to be working.  I'm going to leave this open though so we can check whether this is still necessary in the future.

**Comment by vrothberg** [comment]
Ubuntu 22.04 ships Podman v3.4 (see https://packages.ubuntu.com/search?keywords=podman&searchon=names&suite=jammy&section=all).

@siretart may understand what's going on.  Note that it's best to open issues or bugzillas against the specific Linux distribution as there is not much we can do upstream for dowstream packaging issues.

**Comment by siretart** [comment]
> The podman team closed this issue and referred me to the Debian package managers. I'm not quite sure how to submit an issue there and I need to move on, so I'm going to leave this be.

Please follow the instructions at https://www.debian.org/Bugs/Reporting for reporting bugs to debian. I highly recommend using the `reportbug` tool as instructed on that page.

As for the specific error, I'm unable to reproduce this issue with `podman run -it --rm ubuntu:22.04 bash -c 'apt update && apt install -y podman'` -- that command passes for me just fine.

The error output is probably missing the relevant part that woul indicate why the package manager might be having difficulties.

The cases where `--force-overwrite` is helping indicates a bug in involved pacakges. Most of the time I help users with these symptoms it turns out they were using poorly maintained packages from non-official archives. I'd advise sticking with official packaging repositories -- they tend to undergo stricter QA checks and have higher quality.

**Comment by jefflill** [comment]
**FIXED:** We upgraded to Ubuntu 22.04 and where Podman is on the official mirrors now.
